ocaml/database/db_cache.ml

```diff
@@ -48,4 +53,11 @@ let apply_delta_to_cache entry db_ref =
   | Redo_log.WriteField (tblname, objref, fldname, newval) ->
       debug "Redoing write_field %s (%s) [%s -> %s]" tblname objref fldname
         newval ;
-      DB.write_field db_ref tblname objref fldname newval
+      let removed =
+        try lifecycle_state_of ~obj:tblname fldname = Removed_s
+        with Not_found ->
+          D.warn "no lifetime information about %s.%s, ignoring" tblname fldname ;
+          true
+      in
+      if not removed then
```

Comment:
It would be useful to add a log line if `removed = true`, so that we can see what is going on, and we don't assume that the `Redoing write_field ...` has actually happened.